File: tier1/csharp/418521bb/Report.cs
using System.Collections.Generic;
using System.Diagnostics;

public static class Report
{
    private static readonly Dictionary<string, string> Allowed =
        new Dictionary<string, string> { { "daily", "daily" }, { "weekly", "weekly" } };

    public static string Build(string name)
    {
        if (!Allowed.TryGetValue(name, out var target))
        {
            return "";
        }
        return Run(target);
    }

    private static string Run(string target)
    {
        Process.Start("/usr/bin/report", target);
        return target;
    }
}

File: tier1/java/3d75e894/Report.java
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public final class Report {
    private static final Map<String, String> ALLOWED = new HashMap<>();
    static {
        ALLOWED.put("daily", "daily");
        ALLOWED.put("weekly", "weekly");
    }

    public static String build(String name) {
        String target = ALLOWED.get(name);
        if (target == null) {
            return "";
        }
        return run(target);
    }

    private static String run(String target) {
        try {
            Runtime.getRuntime().exec("/usr/bin/report " + target);
        } catch (IOException e) {
            return "";
        }
        return target;
    }
}
